### ISSUE TITLE
Unread count goes from zero through the roof after restarting the app

### ISSUE THREAD

**Issue by Johennes** [vector-im/element-web#25479]
### Steps to reproduce

1. Go to inbox zero in https://matrix.to/#/#ps-features-team:matrix.org
2. Restart and relaunch Element Desktop
3. Open the room
4. Notice how the unread count slowly jumps up to ~500

#### What did you expect?

Lots of messages are marked unread

#### What happened instead?

No unread count in the room

### Operating system

macOS

### Application version

Element Desktop nightly

### Homeserver

element.io

### Will you send logs?

Yes

**Comment by turt2live** [comment]
I've noticed this primarily on DMs, ftr

**Comment by Johennes** [comment]
Can't reproduce the rapidly rising unread count anymore as of this morning. Travis had posted to the main timeline in the room to clear his stuck notification which might be related.

**Comment by weeman1337** [comment]
I can also see this problem from time to time. That happens rarely.

**Comment by gsouquet** [comment]
This issue is a duplicate of https://github.com/vector-im/element-web/issues/25450

Digging through the logs I can find the following exception

```
2023-05-30T14:48:53.690Z E Error sending receipt {"room":"!pDXDyxlzCKaaGYESIR:matrix.org","error":{"httpStatus":400,"url":"https://element.ems.host/_matrix/client/r0/rooms/!pDXDyxlzCKaaGYESIR%3Amatrix.org/receipt/m.read/%24JgIrwa6haKY-PoeAGUzyzR3Ceii9ywNLl_057T39dPg","errcode":"M_INVALID_PARAM","data":{"errcode":"M_INVALID_PARAM","error":"event_id $JgIrwa6haKY-PoeAGUzyzR3Ceii9ywNLl_057T39dPg is not related to thread main"},"name":"M_INVALID_PARAM"}}
```

The last event in the main live timeline is a reaction that belongs to a thread. Your accumulated read receipt refers to an older event. 
What happens is that the client tries to send a read receipt agains the latest event in the live timeline, but it should not have ended up there.
The client eagerly clears the notification count, but the API call fails, and this failure mode is not taken into account... which means that upon a restart, your accumulated read receipt and the server read receipt has not changed because the API call has failed, you're therefore replaying the notification.

**Comment by turt2live** [comment]
@gsouquet I've definitely seen this happen when it's not a reaction in the main timeline. It appears others have as well (rageshakes hopefully inbound on this).

**Comment by weeman1337** [comment]
If this is a release blocker and a duplicate of https://github.com/vector-im/element-web/issues/25450, should the other issue become a release blocker?

**Comment by Johennes** [comment]
Yes, good catch. Have made it so. We've internally been treating it like this already.

**Comment by Johennes** [comment]
I'm still seeing this on nightly 2023060102 which includes https://github.com/matrix-org/matrix-js-sdk/pull/3427. Have logged https://github.com/matrix-org/element-web-rageshakes/issues/21713. Reopening this but taking off the release blocker label because the general stuck notification experience is much better on this version.

**Comment by nadonomy** [comment]
If it helps, I'm experiencing this when clicking on 'mark as read'. It's happened to me twice in the last few days but I wasn't able to rageshake. Will do so next time.

**Comment by davidegirardi** [comment]
The latesst Element Nightly 2023062901 seems to go through the root and back to zero by itself, it it helps.

**Comment by aurorasmiles** [comment]
Not sure if it's actually the same issue, but I've started experiencing this randomly on Element Web v1.11.35 every other day now. I click on a room that has an (actual) unread message, and sometimes it just starts marking thee last couple hundred messages or so as unread in that room

**Comment by Johennes** [comment]
Yes, that sounds exactly like what I'm seeing, too

**Comment by thibaultamartin** [comment]
There is a reproducer in https://github.com/matrix-org/element-web-rageshakes/issues/22619, with videos to show it in action, and notification debug information for the room where the issue is happening

**Comment by colans** [comment]
> If it helps, I'm experiencing this when clicking on 'mark as read'. It's happened to me twice in the last few days but I wasn't able to rageshake. Will do so next time.

See https://github.com/vector-im/element-web/issues/26475 for this; I just created it.

**Comment by davidegirardi** [comment]
I can reproduce this consistently with Element Nightly version: 2023112101 Crypto version: Olm 3.2.15.

Steps:
- Go to an active room with threads
- Make sure your right panel is closed
- Restart or refresh Element Nightly
- Open a thread from the timeline which will open the panel on the right
- Read the thread
- Click on the left arrow at the top of the threads side panel to show all the threads
- Look at the notification count skyrocket while Element renders events like `/confetti` or 🎉  at the same time

Bonus: scroll up to get even more unreads

**Comment by salvador-diaz** [comment]
I'm experiencing the same issue as described by _davidegirardi_ https://github.com/element-hq/element-web/issues/25479#issuecomment-1822286616 with Element version: 1.11.51 Crypto version: Olm 3.2.15

**Comment by davidegirardi** [comment]
@salvador-diaz that's an ancient version (with known security vulnerabilities no less). Please upgrade.

I haven't encountered this behaviour in months.

**Comment by salvador-diaz** [comment]
@davidegirardi my bad 😅️, upgraded to 1.11.79 and problem solved. Thanks
